src/error/MatterError.ts

```diff
@@ -0,0 +1,146 @@
+
+export class UnreportableAttributeStatusResponseError extends StatusResponseError {}
+StatusResponseError.registerErrorClass(StatusCode.UnreportableAttribute, UnreportableAttributeStatusResponseError);
+
+export class InvalidDataTypeStatusResponseError extends StatusResponseError {}
+StatusResponseError.registerErrorClass(StatusCode.InvalidDataType, InvalidDataTypeStatusResponseError);
+
+export class UnsupportedReadStatusResponseError extends StatusResponseError {}
+StatusResponseError.registerErrorClass(StatusCode.UnsupportedRead, UnsupportedReadStatusResponseError);
+
+export class DataVersionMismatchStatusResponseError extends StatusResponseError {}
+StatusResponseError.registerErrorClass(StatusCode.DataVersionMismatch, DataVersionMismatchStatusResponseError);
+
+export class TimeoutStatusResponseError extends StatusResponseError {}
+StatusResponseError.registerErrorClass(StatusCode.Timeout, TimeoutStatusResponseError);
+
+export class UnsupportedModeStatusResponseError extends StatusResponseError {}
+StatusResponseError.registerErrorClass(StatusCode.UnsupportedMode, UnsupportedModeStatusResponseError);
+
+export class BusyStatusResponseError extends StatusResponseError {}
+StatusResponseError.registerErrorClass(StatusCode.Busy, BusyStatusResponseError);
+
+export class UnsupportedClusterStatusResponseError extends StatusResponseError {}
+StatusResponseError.registerErrorClass(StatusCode.UnsupportedCluster, UnsupportedClusterStatusResponseError);
+
+export class NoUpstreamSubscriptionStatusResponseError extends StatusResponseError {}
+StatusResponseError.registerErrorClass(StatusCode.NoUpstreamSubscription, NoUpstreamSubscriptionStatusResponseError);
+
+export class NeedsTimedInteractionStatusResponseError extends StatusResponseError {}
+StatusResponseError.registerErrorClass(StatusCode.NeedsTimedInteraction, NeedsTimedInteractionStatusResponseError);
+
+export class UnsupportedEventStatusResponseError extends StatusResponseError {}
+StatusResponseError.registerErrorClass(StatusCode.UnsupportedEvent, UnsupportedEventStatusResponseError);
+
+export class PathsExhaustedStatusResponseError extends StatusResponseError {}
+StatusResponseError.registerErrorClass(StatusCode.PathsExhausted, PathsExhaustedStatusResponseError);
+
+export class TimedRequestMismatchStatusResponseError extends StatusResponseError {}
+StatusResponseError.registerErrorClass(StatusCode.TimedRequestMismatch, TimedRequestMismatchStatusResponseError);
+
+export class FailsafeRequiredStatusResponseError extends StatusResponseError {}
+StatusResponseError.registerErrorClass(StatusCode.FailsafeRequired, FailsafeRequiredStatusResponseError);
+
+
+export function isMatterError(e: unknown): e is MatterError {
+    return e instanceof MatterError;
+}
+
+type MatterErrorHandler<T> = (error: MatterError) => T | undefined;
+
+export function tryCatch<T>(codeBlock: () => T, errorType: typeof MatterError, fallbackValue: MatterErrorHandler<T> | T): T {
```

Comment:
nit: fallbackValue -> fallback since this can be a function or a value ?

Reply:
I renamed to fallbackValueOrFunction :-)